eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java

```diff
@@ -92,7 +94,8 @@ public void init() {
             .setTracerProvider(sdkTracerProvider)
             .build();
 
-        Runtime.getRuntime().addShutdownHook(new Thread(sdkTracerProvider::close));
+        shutdownHook = new Thread(sdkTracerProvider::close);
+        Runtime.getRuntime().addShutdownHook(shutdownHook);
```

Comment:
I don't think this change is needed.

Reply:
^ to keep the reference to the shutdownHookThread, thus the behavior of `addShutdownHook` could be tested (it raise `Hook previously registered` when the same shutdownHookThread is added at second time).

https://github.com/apache/incubator-eventmesh/blob/6ea8944b8c91314b296acbb9ef107902c32b084d/eventmesh-trace-plugin/eventmesh-trace-zipkin/src/test/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceServiceTest.java#L47-L55

the `new Thread(sdkTracerProvider::close)` is not the same shutdownHookThread as previous, it could be added successfully and doesn't raise any exception (thus the behavior of `addShutdownHook` could not be tested).

https://github.com/apache/incubator-eventmesh/blob/836e3a5bd6a25c9831b0c6a8db30c997c68f0e91/eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java#L95

Reply:
In your assert, you add a shutdown hook twice, then it throws an exception. Does this test make sense?

Reply:
> add a shutdown hook twice, then it throws an exception

the shutdown hook is already added in the `zipkinTraceService.init();`, thus it throws exception in test as expected.

It ensures the `addShutdownHook` is called in the `init()` method, and the `Thread(sdkTracerProvider::close)` is added there. If someone change the behavior of that (e.g. remove the whole line `Runtime.getRuntime().addShutdownHook(new Thread(sdkTracerProvider::close));`), the unit test would fail.

Reply:
OK, I got it.